tests/jax/components/building/networks_test.py

```diff
@@ -0,0 +1,209 @@
+                "critic_layer_sizes": critic_layer_sizes,
+                "net_key": net_key,
+            }
+
+        return networks
+
+    def network_factory(*args: Any, **kwargs: Any) -> Any:
+        return make_default_networks(  # type: ignore
+            policy_layer_sizes=(254, 254, 254),
+            critic_layer_sizes=(512, 512, 256),
+            *args,
+            **kwargs,
+        )
+
+    return network_factory
+
+
+@pytest.fixture
+def test_builder() -> SystemBuilder:
+    """Pytest fixture for system builder.
+
+    Adds mock env specs and agent net keys to store.
+
+    Returns:
+        System builder with no components.
+    """
+    test_builder = Builder(components=[])
+    test_builder.store.environment_spec = make_fake_env_specs()
+    test_builder.store.agent_net_keys = {
+        "net_key1": "network_1",
+        "net_key2": "network_2",
+    }
+    return test_builder
+
+
+@pytest.fixture
+def test_default_networks(test_network_factory: Callable) -> Networks:
+    """Pytest fixture for default networks.
+
+    Args:
+        test_network_factory: factory to use in default networks config.
+
+    Returns:
+        Default networks test component.
+    """
+    networks_config = NetworksConfig()
+    networks_config.network_factory = test_network_factory
+    networks_config.seed = 919
+
+    return DefaultNetworks(networks_config)
+
+
+def test_key_in_store(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if key is loaded into the store by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    assert not hasattr(test_builder.store, "key")
+    test_default_networks.on_building_init_start(test_builder)
+    assert hasattr(test_builder.store, "key")
+    assert isinstance(test_builder.store.key, jax.random.PRNGKeyArray) or isinstance(
+        test_builder.store.key, jax.numpy.DeviceArray
+    )
+
+
+def test_config_set(test_default_networks: Networks) -> None:
+    """Test if config is set by component init.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+
+    Returns:
+        None.
+    """
+    assert test_default_networks.config.seed == 919
+
+
+def test_network_factory_environment_spec(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if environment spec is given to the network factory and stored by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    test_default_networks.on_building_init_start(test_builder)
+    networks = test_builder.store.network_factory()
+
+    for network in networks.values():
+        assert network["environment_spec"]._keys == ["agent_0", "agent_1"]
```

Comment:
Can we please get the `["agent_0", "agent_1"]` values from `make_fake_env_specs()` directly. It is possible someone changes the default values in `make_fake_env_specs()` and then this test fails.

Reply:
Done. Thanks for spotting this.